ravens/tasks/planners.py

```diff
@@ -0,0 +1,203 @@
+import abc
+
+import matplotlib.pyplot as plt
+import numpy as np
+from ravens.utils import utils
+from scipy.interpolate import interp1d
+from scipy.spatial.transform import Rotation, RotationSpline
+
+
+class Planner(abc.ABC):
+    """Spline-based planner base class.
+
+    Subclasses must implement `_preprocess_poses` and `_postprocess_poses` methods.
+    """
+
+    def __init__(self, steps_per_seg=2, t_max=10.0, height=0.32):
+        self.steps_per_seg = steps_per_seg
+        self.t_max = t_max
+        self.height = height
+
+        self.poses = None
+        self.rots = None
+        self.rot_spline = None
+        self.pos_spline = None
+
+    @abc.abstractclassmethod
+    def _preprocess_poses(self, *poses):
+        pass
+
+    @abc.abstractclassmethod
+    def _postprocess_poses(self, poses, xnew):
```

Comment:
```suggestion
    def _postprocess_poses(cls, poses, xnew):
```

Should be cls since it's a classmethod.